setup.py

```diff
@@ -38,6 +38,5 @@
                       'gradient_free': ['gradient-free-optimizers']},
       zip_safe=False,
       use_scm_version={'fallback_version': '1.0.0'},
-      setup_requires=['setuptools_scm'],
```

Comment:
I think this is used to match the version to the latest tag, does it not work aymore?

Reply:
I'm not convinced its being run. I tried adding the `wheel` package to this but with no result. The setuptools documentation say that `setup_requires` is now deprecated, so not sure if its still working

Reply:
I also tried unpinning pip but this just gave a new error https://github.com/fitbenchmarking/fitbenchmarking/runs/4455891900?check_suite_focus=true

Reply:
Maybe we should do something like this as seen in the release workflow? https://github.com/fitbenchmarking/fitbenchmarking/blob/bfad0eb754e16f32f1d82c24fd196285efb73726/.github/workflows/release.yml#L21

Or the other option might be to switch to using `pyproject.toml`, but this will take longer to implement

Reply:
Yeah in the long run a swithc to pyproject.toml would be best, but I think that can wait until the new year. Fo now lets just get this passing again! I'll merge this after making some notes on #952